backend/app/routes/moments/deleteMoment.js

```diff
@@ -0,0 +1,73 @@
+import databaseUtil from '../../utility/DatabaseUtil';
+import {
+    GOOD,
+    UNAUTHORIZED,
+    INVALID_INPUT,
+    INVALID_USER_OPERATION,
+    MOMENT_DOES_NOT_EXIST,
+    UNKNOWN_ERROR,
+} from '../../utility/ResponseCodes';
+
+const deleteMoment = {
+    method: 'DELETE',
+    path: '/api/moment/{id}',
+    handler: (request, reply) => {
+        // Check if authorized
+        if (!request.auth.credentials) {
+            return reply
+                .response({ code: UNAUTHORIZED.code })
+                .code(UNAUTHORIZED.http);
+        }
+        // Get the user id
+        let userId = request.auth.credentials.user.id;
+
+        // Get the moment id
+        let momentId = request.params.id;
+
+        // Check if the caption id is valid
+        if (momentId === undefined || momentId === '' || !/^\d+$/.test(momentId)) {
+            return reply
+                .response({ code: INVALID_INPUT.code })
+                .code(INVALID_INPUT.http);
+        }
+
+        // Parse the moment id to an integer
+        userId = parseInt(userId, 10);
+        momentId = parseInt(momentId, 10);
+
+        // First check if the moment exists in the db
+        const checkMoment = 'SELECT * FROM MOMENT WHERE MOMENT.ID=? AND DELETED=0';
```

Comment:
Don't do this, rather delete it straight up and check the mysql errors and dispatch error codes based on the result